bin/www

```diff
@@ -69,8 +80,47 @@ function onError(error) {
 /**
  * Event listener for HTTP server "listening" event.
  */
-function onListening() {
+function onListening(server) {
   const addr = server.address()
   const bind = typeof addr === 'string' ? 'pipe ' + addr : 'port ' + addr.port
   debug('Listening on ' + bind)
 }
+
+/**
+ *
+ * Handles gradelful shutdown
+ *
+ * @param server  - The server instance to be shut down.
+ * @param {Object} resources - The resources to be cleaned up during shutdown.
+ */
+function onShutdown(server, { cachingService, logger }) {
+  logger.info('Shutdown started')
+  server.close(() => {
+    logger.info('Server closed')
+    Promise.allSettled([cachingService.done()]).then(results => {
+      const errorResults = results.filter(result => result.status === 'rejected')
+      if (errorResults.length === 0) {
+        logger.info('Shutdown complete')
+        process.exit(0)
+      }
+      errorResults.forEach(errorResult => logger.error(errorResult.reason))
+      process.exit(1)
+    })
+  })
+}
+
+/**
+ * Connect to the required services and return the resources
+ *
+ * @param {Object} config - The configuration object
+ * @returns {Promise<Object>} - The resources object
+ */
+async function connect(config) {
+  const logger = loggerFactory(config.logging.logger())
+  const cachingService = config.caching.service()
+  await Promise.all([cachingService.initialize()]).catch(error => {
+    logger.error('Error connecting to cachingService', error)
+    process.exit(1)
```

Comment:
Fail fast here when redis fails to connect during startup. In the event of this happens, we have the option to update configuration to use in-memory caching for deployment as a fallback.

Reply:
Problem is that while you wait for connection, your service is not exposing any endpoint. How do you deal with the cloud probes ?